src/Symfony/Component/Finder/Expression/Glob.php

```diff
@@ -100,58 +102,8 @@ public function isExpandable()
      */
     public function toRegex($strictLeadingDot = true, $strictWildcardSlash = true)
     {
-        $firstByte = true;
-        $escaping = false;
-        $inCurlies = 0;
-        $regex = '';
-        $sizeGlob = strlen($this->pattern);
-        for ($i = 0; $i < $sizeGlob; $i++) {
-            $car = $this->pattern[$i];
-            if ($firstByte) {
-                if ($strictLeadingDot && '.' !== $car) {
-                    $regex .= '(?=[^\.])';
-                }
-
-                $firstByte = false;
-            }
-
-            if ('/' === $car) {
-                $firstByte = true;
-            }
-
-            if ('.' === $car || '(' === $car || ')' === $car || '|' === $car || '+' === $car || '^' === $car || '$' === $car) {
-                $regex .= "\\$car";
-            } elseif ('*' === $car) {
-                $regex .= $escaping ? '\\*' : ($strictWildcardSlash ? '[^/]*' : '.*');
-            } elseif ('?' === $car) {
-                $regex .= $escaping ? '\\?' : ($strictWildcardSlash ? '[^/]' : '.');
-            } elseif ('{' === $car) {
-                $regex .= $escaping ? '\\{' : '(';
-                if (!$escaping) {
-                    ++$inCurlies;
-                }
-            } elseif ('}' === $car && $inCurlies) {
-                $regex .= $escaping ? '}' : ')';
-                if (!$escaping) {
-                    --$inCurlies;
-                }
-            } elseif (',' === $car && $inCurlies) {
-                $regex .= $escaping ? ',' : '|';
-            } elseif ('\\' === $car) {
-                if ($escaping) {
-                    $regex .= '\\\\';
-                    $escaping = false;
-                } else {
-                    $escaping = true;
-                }
-
-                continue;
-            } else {
-                $regex .= $car;
-            }
-            $escaping = false;
-        }
-
-        return new Regex('^'.$regex.'$');
+        $regex = FinderGlob::toRegex($this->pattern, $strictLeadingDot, $strictWildcardSlash);
+
+        return new Regex(substr($regex, 1, -1));
```

Comment:
well it look like a hack, why not adding new argument in [toRegex](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Finder/Glob.php#L47) ?

```php
public static function toRegex(..., $delimiter = '#')
{
    ....
}